crates/aiken-lang/src/tipo/expr.rs

```diff
@@ -2444,3 +2456,44 @@ pub fn ensure_serialisable(allow_fn: bool, t: Rc<Type>, location: Span) -> Resul
         },
     }
 }
+
+fn is_pattern_expectable(pattern: &TypedPattern, is_cast: bool, level: usize) -> bool {
+    match pattern {
+        Pattern::Int { .. } | Pattern::Discard { .. } => true,
+        Pattern::Var { .. } => !is_cast || level != 0,
```

Comment:
Not sure to understand the rationale with this `level` here 🤨

Reply:
It allows Some(opaque_type_as_var)